app-admin/etcd-wrapper/files/etcd-wrapper

```diff
@@ -4,6 +4,11 @@
 # ExecStart here.
 set -e
 
+# Since etcd/v3 we can't use both `--name` and `ETCD_NAME` at the same time.
+# We parse the etcd command line options to find a `--name` if we found one,
+# we unset the `ETCD_NAME` to not conflict with it.
+[[ $(echo ${@} | grep "\-\-name") ]] && unset ETCD_NAME
```

Comment:
This would miss a case when someone uses just one dash (`-name`) instead of two dashes (`--name`). But grepping for a single dashed option would also match flags like `-discovery-srv-name`. I'd suggest iterating the `${@}` array:
```bash
for f; do
    if [[ $f =~ ^-?-name=? ]]; then
        unset ETCD_NAME
        break
    fi
done
```

If someone passes something like `--discovery-srv-name --name` where `--name` is the name used for DNS discovery then eh, self inflicted pain I'd say.

Reply:
yeah right, I tend to forget flag passed with only one dash (`-name`) - I wanted to avoid iterating over `${@}` but let's do it then. :)

Reply:
Done in a following force pushed commit.